src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -250,21 +262,36 @@ public void Reset()
 
         protected void ResetComponents()
         {
-            var frameHeaders = Interlocked.Exchange(ref _frameHeaders, null);
-            if (frameHeaders != null)
+            if(ServerOptions?.CacheHeadersAndStreams ?? false)
```

Comment:
This is null in some of the tests so it fails.  Let me know if changing tests is a better option.